reinforced_lib/agents/thompson_sampling.py

```diff
@@ -43,9 +29,9 @@ class AgentState(NamedTuple):
     last_decay: chex.Array
 
 
-def thompson_sampling(context: chex.Array, decay: jnp.float32 = 1.0) -> BaseAgent:
+def thompson_sampling(context: chex.Array, decay: chex.Scalar = 1.0) -> BaseAgent:
```

Comment:
Czemu to jest jako funkcja, a nie jako klasa która rozszerza BaseAgent? W sensie trochę nie czaję tej struktury, nie mówie że to jest niedobre.

Reply:
Ponieważ programujemy funkcyjnie w JAX i nasz agent to tylko zestaw funkcji, nie ma żadnego stanu. Stan przechowujemy gdzieś dla każdej instancji jako osobna zmienna, natomiast agenta tworzymy tylko raz i on obsługuje wszystkie instancje. To ma dodatkową zaletę, że funkcje kompilują się tylko raz, a nie za każdym razem gdy tworzymy nowy obiekt (wystarczy stworzyć tylko stan agenta przez `init`).

Reply:
Jasne, wszystko zrozumiałe. Zapomiałem że chcemy wykorzystać ten _jax.jit_